rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,378 @@
+        self.policy = FCNet(config)
+
+    def input_specs(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+                "actions": TorchTensorSpec("b"),
+            }
+        )
+
+    def output_specs(self) -> ModelSpec:
+        return ModelSpec({"action_dist": Distribution})
+
+    def input_specs_exploration(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_inference(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_train(self) -> ModelSpec:
+        return self.input_specs()
+
+    def output_specs_exploration(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_inference(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_train(self) -> ModelSpec:
+        return self.output_specs()
+
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        with torch.no_grad():
+            action_logits = self.policy(obs)
+        action_logits_inference = torch.argmax(action_logits, dim=-1)
+        action_dist = TorchDeterministic(action_logits_inference)
+        return {"action_dist": action_dist}
+
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        with torch.no_grad():
+            return self._forward_train(batch)
+
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_dist = TorchCategorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    def is_distributed(self) -> bool:
+        return False
+
+    def make_distributed(self, dist_config: Mapping[str, Any] = None) -> None:
+        pass
+
+    def get_state(self) -> Mapping[str, Any]:
+        return {"policy": self.policy.state_dict()}
+
+    def set_state(self, state: Mapping[str, Any]) -> None:
+        self.policy.load_state_dict(state["policy"])
+
+
+class BCTorchOptimizer(RLOptimizer):
+    def __init__(self, module, config):
+        super().__init__(module, config)
+
+    def compute_loss(
+        self, batch: NestedDict, fwd_out: Mapping[str, Any]
+    ) -> Mapping[str, Any]:
+        """Compute a loss"""
+        action_dist = fwd_out["action_dist"]
+        loss = -action_dist.logp(batch[SampleBatch.ACTIONS]).mean()
+        loss_dict = {
+            "total_loss": loss,
+        }
+        return loss_dict
+
+    def _configure_optimizers(self) -> List[torch.optim.Optimizer]:
+        return {
+            "module": torch.optim.SGD(
```

Comment:
shouldn't this key be `total_loss` based on what we discussed?

Reply:
wait but how would you use these keys here? where would you `module` for example?

Reply:
resolved offline.